src/d/actor/d_a_obj_shelf.cpp

```diff
@@ -4,133 +4,288 @@
 //
 
 #include "d/actor/d_a_obj_shelf.h"
+#include "d/res/res_otana.h"
 #include "d/d_procname.h"
+#include "d/d_com_inf_game.h"
+#include "d/actor/d_a_npc_ba1.h"
+#include "m_Do/m_Do_mtx.h"
+
+#include "weak_data_1811.h" // IWYU pragma: keep
+
+Mtx daObjShelf::Act_c::M_tmp_mtx;
+
+namespace daObjShelf {
+    namespace {
+        struct Attr_c {
+            /* 0x00 */ short mVibDuration;
+            /* 0x02 */ short mVibYSpeed;
+            /* 0x04 */ float mVibYMagnitude;
+            /* 0x08 */ short mVibXSpeed;
+            /* 0x0a */ short mVibZSpeed;
+            /* 0x0c */ float mVibXMagnitude;
+            /* 0x10 */ float mVibZMagnitude;
+            /* 0x14 */ float mRotAccel;
+            /* 0x18 */ float mRotDecay;
+            /* 0x1c */ float mBounceFactor;
+            /* 0x20 */ u8 mBounceNum;
+            /* 0x21 */ u8 mRotWaitDuration;
+            /* 0x22 */ short m22;
+            /* 0x24 */ float mInitSpeed2;
+            /* 0x28 */ float mInitSpeed3;
+        };
+
+        static const Attr_c L_attr = {
+            4,
+            30000, 4.0f,
+            0x7724,0x7148,
+            200.0f,200.0f,
+            600.0f,0.02f,
+            0.5f,
+            5,0,
+            0,
+            -3500.0f,-2500.0f
+        };
+
+        inline const Attr_c & attr() { return L_attr; }
+    }
+}
+
+const char daObjShelf::Act_c::M_arcname[] = "Otana";
 
 /* 00000078-0000012C       .text CreateHeap__Q210daObjShelf5Act_cFv */
-void daObjShelf::Act_c::CreateHeap() {
-    /* Nonmatching */
+int daObjShelf::Act_c::CreateHeap() {
+    J3DModelData* model_data = (J3DModelData *)dComIfG_getObjectRes(M_arcname, OTANA_BDL_OTANA);
+    JUT_ASSERT(0x12c, model_data != 0);
```

Comment:
```suggestion
    JUT_ASSERT(0x12c, model_data != NULL);
```
It appears as "0" in the string, but the programmers likely wrote NULL and the compiler converted it like that due to nested macros.